### PR TITLE
added model resources for xlm-roberta

### PR DESCRIPTION
**Pull request by hazrulakmal**
# What does this PR do?

Fixes [20055](https://github.com/huggingface/transformers/issues/20055)

- I created a link to task guides for casual language modeling and text classification. I think they are useful and applicable but not directly related to the xlm-roberta model class per say. 
-  For casual language modeling, should I write it under "text-generation" pipeline tag or create a subheader like multiple choice?
- I've checked notebooks from the community but so far none of them do a tutorial on xlm-roberta. Hopefully, there'll be one soon!
- I've also found a few blog posts related to roberta but not xlm-roberta. should we include them? cus they are technically the same architecture just that one is multi-lingual and the other is not

## Before submitting
- [x] This PR improves the docs of xlm-roberta by adding common and most used resources
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).

@stevhliu please check the work and let me know if I need to do any changes. thanks

### CODE REVIEW

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 1]
```diff
@@ -43,6 +43,53 @@ Tips:
...
+- [`FlaxXLMRobertatForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
+- [Token classification](https://huggingface.co/course/chapter7/2?fw=pt) chapter of the 🤗 Hugging Face Course.
+
+<PipelineTag pipeline="text-generation"/>
+
+- [`XLMRobertaForCasualLM`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling.ipynb).
+- [`TFXLMRobertaForCasualLM`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/language-modeling#run_clmpy) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling-tf.ipynb).
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.mdx:67]: We don't have a TensorFlow or Flax equivalent for `XLMRobertaForCausalLM` so those two can be removed.

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 2]
```diff
@@ -43,6 +43,53 @@ Tips:
...
+- [`TFXLMRobertaForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
+- [`FlaxXLMRobertatForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
+- [Token classification](https://huggingface.co/course/chapter7/2?fw=pt) chapter of the 🤗 Hugging Face Course.
+
+<PipelineTag pipeline="text-generation"/>
+
+- [`XLMRobertaForCasualLM`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling.ipynb).
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.mdx:66]:
```suggestion
- [`XLMRobertaForCausalLM`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling.ipynb).
```

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 3]
```diff
@@ -43,6 +43,53 @@ Tips:
...
+- [`TFXLMRobertaForCasualLM`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/language-modeling#run_clmpy) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/language_modeling-tf.ipynb).
+- [`FlaxXLMRobertatForCasualLM`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/language-modeling#causal-language-modeling) 
+- [Casual language modeling](https://huggingface.co/docs/transformers/tasks/language_modeling) chapter of the 🤗 Hugging Face Task Guides.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.mdx:69]:
```suggestion
- [Causal language modeling](https://huggingface.co/docs/transformers/tasks/language_modeling#causal-language-modeling) chapter of the 🤗 Hugging Face Task Guides. 
```

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 4]
```diff
@@ -43,6 +43,53 @@ Tips:
 
 This model was contributed by [stefan-it](https://huggingface.co/stefan-it). The original code can be found [here](https://github.com/pytorch/fairseq/tree/master/examples/xlmr).
 
+## Resources
+
+A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with XLM-RoBERTa. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+
+<PipelineTag pipeline="text-classification"/>
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.mdx:50]: There is also a blog post on how to finetune XLMRoberta for text classification with Habana Gaudi [here](https://www.philschmid.de/habana-distributed-training).

**Review by stevhliu — COMMENTED** [review]
Thanks so much for your contribution!

The `text-generation` pipeline is ok and maybe let's keep it strictly to XLM-Roberta for now :)

File: docs/source/en/model_doc/xlm-roberta.mdx (hunk 1 above)
Comment (hazrulakmal) [reply docs/source/en/model_doc/xlm-roberta.mdx]: ops I think I might have done something weird here. I rebased my branch before making the changes that you suggested and now I don't know why suddenly there were commits from other people in this pull request.

File: docs/source/en/model_doc/xlm-roberta.mdx [hunk 5]
```diff
@@ -43,6 +43,52 @@ Tips:
...
+- [Text classification](https://huggingface.co/docs/transformers/tasks/sequence_classification) chapter of the 🤗 Hugging Face Task Guides.
+
+<PipelineTag pipeline="token-classification"/>
+
+- [`XLMRobertaForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification.ipynb).
+- [`TFXLMRobertaForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
+- [`FlaxXLMRobertatForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm-roberta.mdx:62]: Small typo here

```suggestion
- [`FlaxXLMRobertaForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
```

**Review by stevhliu — COMMENTED** [review]
Thanks for the changes, just one last typo left!

Did you `force push` after rebasing? In any case, it might be easier just to close this PR and open a fresh one.

File: docs/source/en/model_doc/xlm-roberta.mdx (hunk 5 above)
Comment (hazrulakmal) [reply docs/source/en/model_doc/xlm-roberta.mdx:62]: Ok! will change that and open a new PR.

**Pull request closed without merging**